### PR TITLE
[script][shape] - Updates, enhancements, streamlining

### PR DESCRIPTION
**Pull request by Raykyn55**
Follow on from previous PR, now that it's working, cleaned it up and updated it. Needs testing with arrow making.

removed includes, added prefixes.

removed get_item/stow_item, replaced most of that with a method to stow tools only if you don't need them (so no more stow/get shaper when tool repeats). 

consolidated the book portions, split them by chapter (enhancements, arrows, everything else). Expanded shape_items to accomidate these additional methods (cable/lighten/laminate)

made stamp a condition inside finish, that also doesn't try to stamp when doing arrows or enhancements.

added a finish option to hold. This is to allow crafters who want to just let the script make them a bow, without trying to stow or log it.

### CODE REVIEW

File: shape.lic [hunk 1]
```diff
@@ -46,108 +42,70 @@ class Shape
     args = parse_args(arg_definitions)
 
     @finish = args.finish
-    @chapter = args.chapter
-    @recipe_name = args.recipe_name
+    @enhancement = Hash[args.laminate.nil? => 'bow lamination', args.lighten.nil? => 'bow lightening', args.cable.nil? => 'bow cable-backing']
+    @recipe_name = args.recipe_name == nil ? (@enhancement[false]) : args.recipe_name
```
Comment (asechrest) [inline shape.lic:46]: These two lines were especially tough to follow for me. Can we simplify these, avoid creating an instance variable when we're using it just to assist creation of another variable on the next line, and avoid the double-negative-ish nature of `(@enhancement[false])`?

**Comment by Raykyn55** [comment]
In response to asechrest, shrunk the args, made it easier to follow and removed my. . .creative use of the hash.

**Comment by Raykyn55** [comment]
Arrow making was worse than I remembered. It appears to be working now.

Arrowheads and shafts added.

**Comment by MahtraDR** [comment]
Resolves https://github.com/rpherbig/dr-scripts/issues/2761

**Comment by Raykyn55** [comment]
Resolves https://github.com/rpherbig/dr-scripts/issues/4324

**Comment by asechrest** [comment]
@Raykyn55 Running well for you? If so, thinking merge and monitor Discord for any complaints.

**Comment by Raykyn55** [comment]
> @Raykyn55 Running well for you? If so, thinking merge and monitor Discord for any complaints.

Let me verify there arent any uncomitted chamges, a few mins.

**Comment by Raykyn55** [comment]
Looks good, just changed echo to DRC.message in a few places, forgot to commit.

**Pull request merged**